app/controllers/companies_controller.rb

```diff
@@ -1,5 +1,7 @@
 class CompaniesController < ApplicationController
   def index
-    @companies = Company.all
+    companies = Company.all
+    @supporter_companies = companies.select(&:supporter?).sort_by.with_index { |company, i| [-company.contribution_count, i] }
```

Comment:
How about something like this:

```
@supporter_companies = Company.fetch_supporters
@other_companies = Company.fetch_non_supporters

# in company.rb

def fetch_supporters
  COMPANIES.select(&:supporter?).sort_by ...
end

def fetch_non_supporters
  COMPANIES.reject(&:supporter?).sort_by ...
end
```
Or to avoid complicating code that'd allow chaining methods:

```
@supporter_companies = Company.supporter.sort_by_supporter_ranking
@other_companies = Company.non_supporter.sort_by_updated_on
```

Current code is confusing:

* sorting is done both in controller and model
* `supporter_companies` are actually sorted twice, first by `all` method, then here
* sorting in controller is unclear (it took me a few minutes to understand why `with_index` is there)

Reply:
@arnvald implemented suggested refactoring. `supporter_companies` are further sorted by decreasing contributor count, and putting all sorting logic in the model keeps it in one place 👍 

Added the rationale for `with_index` (to maintain stable sort order) into commit message :)